cypress/support/commands.js

```diff
@@ -23,3 +24,31 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+Cypress.Commands.add('login', () => {
+  cy.visit('/');
+  cy.get('.navbar-nav a:contains(Login)').click();
+  cy.get('#email').type(Cypress.env('user'));
+  cy.get('#password').type(Cypress.env('password'));
+  cy.get('button[type="submit"]').click();
+})
+
+Cypress.Commands.add('createsANote', data => {
+  cy.contains('Create a new note', { timeout: 30000 }).should('be.visible').click();
+  cy.get('#content').type(data.content);
+  cy.contains('Create').click();
+  cy.get('.list-group').should('contain', data.content);
+})
+
+Cypress.Commands.add('editsANote', data => {
+  cy.get('.list-group').contains(data.content).click();
```

Comment:
```suggestion
  cy.get('.list-group').contains(data.content).should('be.visible').click();
```

Reply:
I suggest that you "always" check that the element is visible before interacting with it to make the tests more robust and close to how a real user uses a web app.

Reply:
Done!